life_expectancy/cleaning.py

```diff
@@ -0,0 +1,136 @@
+    """ Clean the data and return only a given region.
+
+    Args:
+        region: the name of the region to clean
+    """
+    full_data = (
+        data
+        .pipe(_melt_table)
+        .pipe(_clean_year_value_cols)
+        .pipe(_clean_region_col)
+    )
+    region_data = _select_region(full_data, region)
+
+    return region_data
+
+def _load_data(data_path: str) -> pd.DataFrame:
+    """Load the data.
+
+    Returns:
+        data (pd.DataFrame): The data.
+    """
+    return pd.read_table(f'{data_path}/eu_life_expectancy_raw.tsv')
+
+def _melt_table(data: pd.DataFrame) -> pd.DataFrame:
+    """Melt the table to long format.
+    
+    Args:
+        data (pd.DataFrame): The data.
+        
+    Returns:
+        data (pd.DataFrame): The data in long format.
+    """
+
+    # clean column names
+    data[['unit', 'sex', 'age', 'region']] = data['unit,sex,age,geo\\time'].str.split(
+        ',', expand=True
+    )
+    data.drop('unit,sex,age,geo\\time', axis=1, inplace=True)
+
+    # melt table
+    return data.melt(
+        id_vars=['unit', 'sex', 'age', 'region'],
+        var_name='year', value_name='value'
+    )
+
+def _clean_year_value_cols(data: pd.DataFrame) -> pd.DataFrame:
+    """Clean the year and value columns, ensure they are of the correct type and
+        drop rows with missing values.
+
+    Args:
+        data (pd.DataFrame): The data.
+
+    Returns:
+        data (pd.DataFrame): The data with cleaned year and value columns.
+    """
+    # Ensure year is an integer
+    data['year'] = data['year'].str.extract(
+        r'(\d+)', expand=False
+        ).astype(int)
+
+    # Ensure value is a float
+    data['value'] = data['value'].str.extract(
+        r'(\d+.\d+)', expand=False
+        ).astype(float)
+
+    # drop rows with missing values
+    data.dropna(inplace=True)
+
+    return data
+
+def _clean_region_col(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Clean the region column of the data.
+
+    Args:
+        data: The data to clean.
+
+    Returns:
+        The cleaned data.
+    """
+    data['region'] = data['region'].apply(
+        lambda region: region.split('_')[0]
+        )
+
+    return data
+
+def _select_region(data: pd.DataFrame, region: str) -> pd.DataFrame:
+    """Select the region.
+
+    Args:
+        data (pd.DataFrame): The data.
+        region (str): The region to select.
+
+    Returns:
+        data (pd.DataFrame): The data with only the specified region.
+
+    Raises:
+        ValueError: If the region is not in the data.
+    """
+    if region not in data['region'].unique():
+        raise ValueError(f"""
+            The region '{region}' is not found in the data.
+            Please choose from {data['region'].unique().tolist()}""")
+    return data[data['region'] == region]
+
+def _save_data_csv(data_path: str, data: pd.DataFrame, region: str) -> None:
+    """Save the data to a csv file.
+
+    Args:
+        data (pd.DataFrame): The data.
```

Comment:
I think that the type hint is also unnecessary here and maybe the other arguments (data_path and region) should be added here

Reply:
1. yes it was the docstring auto gen
2. missed it! thanks!